cordlib/src/main/kotlin/com.luxoft.blockchainlab.corda.hyperledger.indy/service/IndyService.kt

```diff
@@ -23,72 +19,42 @@ import java.io.File
  */
 @CordaService
 class IndyService(services: AppServiceHub) : SingletonSerializeAsToken() {
-
-    private val poolName = "default_pool"
-    private val credentials = """{"key": "key"}"""
-
-    private val config = TestConfigurationsProvider.config(services.myInfo.legalIdentities.first().name.organisation)
-        ?: EmptyConfiguration
-            .ifNot(
-                ConfigurationProperties.fromFileOrNull(File("indyconfig", "indy.properties")),
-                indyuser
-            ) // file with common name if you go for file-based config
-            .ifNot(
-                ConfigurationProperties.fromFileOrNull(
-                    File(
-                        "indyconfig",
-                        "${services.myInfo.legalIdentities.first().name.organisation}.indy.properties"
-                    )
-                ), indyuser
-            )  //  file with node-specific name
-            .ifNot(EnvironmentVariables(), indyuser) // Good for docker-compose, ansible-playbook or similar
-
     private val logger = KotlinLogging.logger {}
 
     val indyUser: IndyUser
 
     init {
-        val walletName = try {
-            config[indyuser.walletName]
-        } catch (e: Exception) {
-            services.myInfo.legalIdentities.first().name.organisation
-        }
-        val walletConfig = SerializationUtils.anyToJSON(WalletConfig(walletName))
+        val nodeName = services.myInfo.legalIdentities.first().name.organisation
 
-        try {
+        val walletName =  ConfigHelper.getWalletName()
+        val walletPassword = ConfigHelper.getWalletPassword()
 
-            Wallet.createWallet(walletConfig, credentials).get()
-        } catch (ex: Exception) {
-            if (getRootCause(ex) !is WalletExistsException) throw ex else logger.debug("Wallet already exists")
-        }
+        val wallet = WalletHelper.getWallet(WalletConfig(walletName), WalletPassword(walletPassword))
```

Comment:
Why do not use WalletConfig, WalletPassword inside the WalletHelper? You don't customize it, thus can provide high-level function with String as arguments.

Reply:
WalletConfig actually has 3 params (at least, 3 known params). By now cordentity is not able to use other parameters (storage type and path), but when someday it will, ConfigHelper will also provide these data and all we're going to do is just add them to WalletConfig constructor invocation.
WalletPassword is just used for documentation. Previously it was named as `key` and its default value was `key` also, so it was saying nothing about its real purpose.

Btw, WalletConfig and WalletPassword are serializable structures - what Indy expects to receive in form of JSON. So, it is natural to use them here.

Reply:
Yes, I got it. I mean that we can incapsulate such transformation in WallerHelper, because it's a helper. We take the params from config, it usually string and we can use it as arguments directly. Just provide multiple extended interface with variable number of arguments